### DISCUSSION TITLE
Latest Release Breaks building on Arm Linux

### DISCUSSION THREAD

**Discussion by kamranjon** [openai/whisper#1048] [Q&A] [answered]
I am getting this error when trying to build: 
```
ERROR: Could not find a version that satisfies the requirement triton>=2.0.0.dev20221202 
(from openai-whisper) (from versions: none)
ERROR: No matching distribution found for triton>=2.0.0.dev20221202
```

which is a result of this code:
```python
requirements = []
if sys.platform.startswith("linux"):
    triton_requirement = "triton>=2.0.0.dev20221202"
    try:
        import re
        import subprocess
        version_line = subprocess.check_output(["nvcc", "--version"]).strip().split(b"\n")[-1]
        major, minor = re.findall(rb"([\d]+)\.([\d]+)", version_line)[0]
        if (int(major), int(minor)) < (11, 4):
            # the last version supporting CUDA < 11.4
            triton_requirement = "triton==2.0.0.dev20221011"
    except (IndexError, OSError, subprocess.SubprocessError):
        pass
    requirements.append(triton_requirement)
```

which appears to dynamically add triton as a dependency based on whether or not you have linux. Unfortunately Triton may not show up as available on specific linux platforms (like arm) - so this seems to fail. 

Is anyone aware of a work around? Sorry if this is the wrong place - but the repository for whisper appears to have issues turned off, so unsure where to report this.

**Comment by abramovi** [comment]
Same here - can't find requirement - triton==2.0.0.dev20221011

**Comment by jongwook** [comment, marked as answer]
Hi, sorry about the inconvenience; do you use CUDA on your ARM platform? Otherwise the package should work without triton, and I should add a check `platform.machine() == "x86_64"` in the setup.py snippet to exclude the triton dependency.

EDIT: I made a fix in #1051. Please try 

```bash
pip uninstall openai-whisper whisper
pip install git+https://github.com/openai/whisper.git 
```

and let me know if the issue persists. Thanks!

**Reply by malcolmmurdock** [reply]
Hi @jongwook The issue appears to be persisting on our sagemaker container, presumably due to CUDA version. We're seeing the following error:

```ERROR: Could not find a version that satisfies the requirement triton==2.0.0.dev20221011 (from openai-whisper) (from versions: 0.4.1, 0.4.2, 1.0.0, 1.1.0, 1.1.1, 2.0.0.dev20221030, 2.0.0.dev20221031, 2.0.0.dev20221101, 2.0.0.dev20221103, 2.0.0.dev20221105, 2.0.0.dev20221117, 2.0.0.dev20221120, 2.0.0.dev20221202, 2.0.0.dev20230208, 2.0.0.dev20230217, 2.0.0a1, 2.0.0a2, 2.0.0)```

Thanks for your help!

**Reply by jongwook** [reply]
@captainnurple Hi, it appears that the version `dev20221011` was removed from PyPI. How easy is it to upgrade the container to one with CUDA 11.6+? I'll check with if it can be revived on PyPI.

**Reply by malcolmmurdock** [reply]
Ah that explains it! I'm not actually sure how easy it will be to upgrade the container. Now that we know this is potentially on the horizon we'll start working on it, but if it's possible to restore on PyPI that would be much appreciated since we have folks using this model in the meantime and it would be nice to not have to try and set up a new container that's pinned to an earlier commit. Thanks for checking 🙏

**Reply by jongwook** [reply]
Sure! Also, you could try installing the previous version of `openai-whisper` from PyPI which did not depend on triton. Triton dependency was added for the word-level timestamp feature, so the old version should work well (and without the regression discussed in #1046)

```
pip install openai-whisper==20230124
```

**Reply by malcolmmurdock** [reply]
Ok good to know, since we're not using word level timestamps. We'll take a look. in the meantime, if it's easy to get the PyPI version restored, that would be amazing since we're dead in the water right now 🤪

thanks again, love all the work you guys are doing... 🙏

**Reply by malcolmmurdock** [reply]
openai-whisper==20230124 resolved our issue for now. Thanks @jongwook 🙏
